crates/driver/src/driver.rs

```diff
@@ -1,393 +0,0 @@
-    /// prepares the auction for the solver.
-    pub async fn on_auction_started(
-        &self,
-        auction: AuctionWithId,
-    ) -> Result<SettlementSummary, ()> {
-        // TODO get deadline from autopilot auction
-        let deadline = Instant::now() + Duration::from_secs(25);
-        Self::solve_until_deadline(
-            auction,
-            self.solver.clone(),
-            self.auction_converter.clone(),
-            self.block_stream.clone(),
-            deadline,
-        )
-        .await
-        .map_err(|_| ())
-    }
-
-    /// Computes a solution with the liquidity collected from a given block.
-    async fn compute_solution_for_block(
-        auction: AuctionWithId,
-        block: BlockInfo,
-        converter: Arc<dyn AuctionConverting>,
-        solver: Arc<dyn CommitRevealSolving>,
-    ) -> Result<SettlementSummary> {
-        let block = block.number;
-        let auction = converter.convert_auction(auction, block).await?;
-        solver.commit(auction).await
-    }
-
-    /// Keeps solving the auction in a loop with the latest known liquidity
-    /// until the `deadline` has been reached or the `block_stream`
-    /// terminates. This function uses a `WatchStream` to get notified about
-    /// new blocks which will start with yielding the current block
-    /// immediately and will skip intermediate blocks if it observed
-    /// multiple blocks while computing a result.
-    async fn solve_until_deadline(
-        auction: AuctionWithId,
-        solver: Arc<dyn CommitRevealSolving>,
-        converter: Arc<dyn AuctionConverting>,
-        block_stream: CurrentBlockStream,
-        deadline: Instant,
-    ) -> Result<SettlementSummary> {
-        let compute_solutions = into_stream(block_stream.clone()).then(|block| {
-            Self::compute_solution_for_block(
-                auction.clone(),
-                block,
-                converter.clone(),
-                solver.clone(),
-            )
-        });
-        let timeout = tokio::time::sleep_until(deadline.into());
-        tokio::pin!(timeout, compute_solutions);
-
-        let mut current_solution = Err(anyhow::anyhow!("reached the deadline without a result"));
-        loop {
-            tokio::select! {
-                new_solution = compute_solutions.next() => {
-                    match new_solution {
-                        Some(result) => {
-                            tracing::debug!(?result, "computed new result");
-                            current_solution = result;
-                        },
-                        None => return current_solution
-                    }
-                },
-                _ = &mut timeout => return current_solution
-            }
-        }
-    }
-
-    /// Validates that the `Settlement` satisfies expected fairness and
-    /// correctness properties.
-    async fn validate_settlement(&self, settlement: Settlement) -> Result<SimulationWithResult> {
```

Comment:
I think this still needs to be ported into the driver. We wanted the driver to verify solutions before submitting them to prevent accidental bad behaviour from the `solver engine`s.

Reply:
(Not in this PR of course).

Reply:
I think simulation is one aspect of `validate_settlement`. We also do some other basic validations like token conservation and checking that internalized transactions are legal (i.e. in the list of accepted tokens). This is needed because not all rules are enforced on-chain, and breaking some of the off-chain rules results in slashing. We wanted to reference driver to protect against these cases.

Might be worth creating an issue to not forget to get back to this.